### ISSUE TITLE
Create new project by cloning a template

### ISSUE THREAD

**Issue by svenefftinge** [gitpod-io/gitpod#5583]
The new project flow, should allow users to pick a template and create a fresh project based on that.

As a result, a user should be able to create a fresh repository with the initial scaffolding provided by the picked template. The repo should be added to the githoster the user chose and live under the org/group the user had chosen during the project creation flow. 

As with imported projects the user get directed to the configuration page, which should already run a prebuild if the `.gitpod.yml` exists.

To be investigated:
 - create projects only based on existing git repos, or can we support running other project creation tools (spring boot, react starter, etc.) with reasonable complexity and effort?

**Comment by JohannesLandgraf** [comment]
FWIW, with the /quickstart section in docs we (in words) followed a similar flow: https://www.gitpod.io/docs/quickstart/ruby-on-rails

**Comment by JanKoehnlein** [comment]
Are there any insights on the effort yet? Given the time until the official launch we should maybe postpone that and remove it from "needed for shipping"

**Comment by gtsiolis** [comment]
Thanks @JanKoehnlein! Moving this to the _Next Iteration_ for now unless we manage put some effort on this for next week.

**Comment by svenefftinge** [comment]
Now, that new users land in the /new flow, we should allow them to pick a template to start fresh projects rather than only importing existing projects. I think it could be cool to design this feature, so that I can use a context URL with an archive or so, that is used. So people can build links to easily start projects based on their own templates.

**Comment by svenefftinge** [comment]
@jldec @JanKoehnlein happy to jump on a call to explore this in more depth.

**Comment by jldec** [comment]
### MVP design ideas
cc: @gtsiolis 

*How about reusing the list of [docs quickstart](https://www.gitpod.io/docs/quickstart) template repos from the website?*

1. Render the quickstart templates in a single-page view with names, icons?, short descriptions? (something nicer than a dropdown, no need for pagination but a simple in-page search would be nice)

2. Use the github template generate feature to create the repo inside the GitHub UI (instead of via API) e.g. https://github.com/gitpod-io/template-golang-cli/generate This provides users with control over the repo name, and avoids silent auth issues. It also avoids cloning the git history unless users ask for it.

3. Somehow bring the repo name/url back into the Gitpod new project flow - not sure how to do this without a form-like input, but even that would not be the worst thing to do IMO.

**Comment by roboquat** [comment]
@jldec: Issue scheduled in the meta team (WIP: 0)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/issues/5583#issuecomment-957916925):

>/schedule


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by jldec** [comment]
Addendum to OP - https://github.com/gitpod-io/website/issues/1194
Re-enable example workspace choice from Try Now and Sign Up buttons

**Comment by svenefftinge** [comment]
> Use the github template generate feature to create the repo inside the GitHub UI (instead of via API) e.g. https://github.com/gitpod-io/template-golang-cli/generate This provides users with control over the repo name, and avoids silent auth issues. It also avoids cloning the git history unless users ask for it.

It would be nice to reuse something indeed, but this would
1) not work for GitLab users
2) kick users out of the project creation flow

Instead, we should have a single page where users can pick the org, name and whether the repo should be public/private. And then generate and initialize the project for them based on an archive.

**Comment by jldec** [comment]
See also #6645

**Comment by jldec** [comment]
/assign @jankeromnes 

Thanks Jan, let's start with a really minimal design as suggested by @svenefftinge 

It would be great to have the mechanics of new repo creation with error handling working so that we can better understand the necessary user interactions, while we work on the UI design in parallel.

- the input template source comes from a selection earlier in the flow
- the org/name of the new repo is derived/proposed based on the input but can be overriden by the user
- the new repo defaults to public but that can also be overridden by the user

**Comment by jankeromnes** [comment]
Thanks @jldec, good idea!

One small note: I believe that there are at least two ways to create a repo based on a template:
1. Use the GitHub template feature ("Use this template" button going to `/generate` URL, e.g. https://github.com/gitpod-io/template-sveltejs/generate )
2. Do something more "manual" and generic based on an archive (e.g. https://github.com/gitpod-io/template-sveltejs/archive/refs/heads/master.zip )

My current understanding is that we'll likely skip 1. for now (GitHub-only and more restrictive) and implement 2. (but try to follow established standards with respect to project name, commit author, commit message, etc)

**Comment by jldec** [comment]
Yes - good point - I had similar intuitions, but after playing with the Vercel UI (see below) I think the "create repo" UX can be quite small & simple. See also [comments](#issuecomment-959407399) above.

![Screenshot 2021-11-11 at 22 53 43](https://user-images.githubusercontent.com/849592/142061442-e5c1dd5d-5bc8-4603-8a22-fa35c5779b15.png)

**Comment by jankeromnes** [comment]
Following yesterday's [discussion](https://gitpod.slack.com/archives/C02EN94AEPL/p1637855390120100) (internal), we've decided to discontinue work on this issue (i.e. choosing a template during Project creation) for now, and instead focus on https://github.com/gitpod-io/gitpod/issues/6903 (i.e. opening archives in Gitpod); https://github.com/gitpod-io/gitpod/issues/6904 (allowing repository creation from the IDE); and https://github.com/gitpod-io/gitpod/issues/6906 (allowing Project creation from the IDE).

**Comment by jankeromnes** [comment]
In case this is helpful in the future, here is my work-in-progress branch implementing a "create a repository from a template repository" feature:

https://github.com/gitpod-io/gitpod/pull/6850

https://github.com/gitpod-io/gitpod/compare/jx/project-templates

https://github.com/gitpod-io/gitpod/commit/6b081ddd0796b9d37c773fec2081b8cea28192c4

It's able to successfully create new repositories from a GitHub or GitLab template, and has a minimal UI hacked into the dashboard:

<img width="616" alt="Screenshot 2021-11-30 at 13 44 24" src="https://user-images.githubusercontent.com/599268/144049574-38fb8ef5-427f-4710-8d80-8da9716180a9.png">
